services/motion/builtin/builtin_test.go

```diff
@@ -1728,189 +1171,722 @@ func TestStoppableMoveFunctions(t *testing.T) {
+
+		err = ms.StopPlan(ctx, motion.StopPlanReq{ComponentName: fakeBase.Name()})
+		test.That(t, err, test.ShouldBeNil)
+
+		ph2, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(ph2), test.ShouldEqual, 1)
+		test.That(t, ph2[0].Plan.ExecutionID, test.ShouldResemble, executionID)
+		test.That(t, len(ph2[0].StatusHistory), test.ShouldEqual, 2)
+		test.That(t, ph2[0].StatusHistory[0].State, test.ShouldEqual, motion.PlanStateStopped)
+		test.That(t, ph2[0].StatusHistory[1].State, test.ShouldEqual, motion.PlanStateInProgress)
+		test.That(t, len(ph2[0].Plan.Steps), test.ShouldNotEqual, 0)
+
+		// Proves that calling StopPlan after the plan has reached a terminal state is idempotent
+		err = ms.StopPlan(ctx, motion.StopPlanReq{ComponentName: fakeBase.Name()})
+		test.That(t, err, test.ShouldBeNil)
+		ph3, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, ph3, test.ShouldResemble, ph2)
+	})
+
+	t.Run("returns error when called with an unknown component", func(t *testing.T) {
+		t.Parallel()
+		injectedMovementSensor, _, _, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      base.Named("non existent base"),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Destination:        geo.NewPoint(0, 0),
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeError, errors.New("resource \"rdk:component:base/non existent base\" not found"))
+		test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+	})
+
+	t.Run("returns error when called with an unknown movement sensor", func(t *testing.T) {
+		t.Parallel()
+		_, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: movementsensor.Named("non existent movement sensor"),
+			Destination:        geo.NewPoint(0, 0),
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		e := "Resource missing from dependencies. Resource: rdk:component:movement_sensor/non existent movement sensor"
+		test.That(t, err, test.ShouldBeError, errors.New(e))
+		test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+	})
+
+	t.Run("returns error when request would require moving more than 5 km", func(t *testing.T) {
+		t.Parallel()
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Destination:        geo.NewPoint(0, 0),
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeError, errors.New("cannot move more than 5 kilometers"))
```

Comment:
In general would prefer to use the existing types and constructor functions for errors (if they exist) in these checks to make the tests more robust to the strings changing.  Is there a reason you don't prefer that?

Reply:
I've made this error a builtin package level variable & refer to that variable in both the test & the implementation. Is that what you prefer?

Reply:
Per our discussion offline lets save doing this until another ticket